runtime/doc/news.txt

```diff
@@ -75,6 +75,9 @@ LSP
   customizing the transformation of an LSP CompletionItem to |complete-items|.
 • |vim.lsp.diagnostic.from()| can be used to convert a list of
   |vim.Diagnostic| objects into their LSP diagnostic representation.
+• |vim.lsp.buf.references()| now handles multiple clients but no longer
+  triggers the global `textDocument/references` handler from
```

Comment:
Why does this affect the handler? And on that note, might this change affect plugins like Telescope/fzf-lua that "wrap" these methods to display them in the respective pickers?

Reply:
I agree, removal of the handler is technically a breaking change.

It looks like it's because the design of `|lsp-handler|` assumes there is only one LSP client involved in a single LSP request (`ctx.client_id` is not plural). As this implementation tries to fire multiple LSP requests, there would be multiple response and thus multiple LSP handler invocations (called asynchronously); however, it doesn't look very straightforward how to combine and merge the results into a single qflist.

Reply:
There is a similar effort in #26507 for lsp.buf.definitions. Ideally we'd want to migrate all lsp.buf.* calls to be able to handle results from multiple servers

Reply:
I don't mind marking this as breaking but it's imho generally grey area because another point of view is that this would be a fix, given that currently using references with multiple clients is broken in that each client overrides the results from the previous client.

Also it's not breaking in the sense that users would get errors - at worst they'll get the stock neovim behavior, instead of one customized by a plugin. (We also went with a `on_list` argument in anticipation of this change)

https://github.com/neovim/neovim/pull/22598#issuecomment-1463367606 also explains a bit more why this is necessary.
The global-handler model is not suited for any actions where results need to be merged from multiple clients and we'll have to do this for most of the lsp.buf. methods. Hover, signature help, all the go-to variants, etc.

Plugins which currently override the handler and rely on that are similarly broken for multiple-clients.
It could be an option that we preserve that behavior if there is only one client for the buffer - but that would imho just be more confusing and drag out getting it fixed properly